lib/spring/application.rb

```diff
@@ -134,6 +138,7 @@ def run
     end
 
     def serve(client)
+      app_started = [false]
```

Comment:
It serves as a "pointer", for passing by reference instead of by value. I am open to other ideas, and I know it's a bit weird.